sympy/integrals/integrals.py

```diff
@@ -444,6 +444,12 @@ def doit(self, **hints):
             function = function.doit(**hints)
         if function.is_zero:
             return S.Zero
+        if len(self.limits) == 1 and len(self.limits[0]) == 3:
```

Comment:
Let's split this logic out into a new method on Integral, like `is_convergent`. We can then call that method here. But it's better to avoid inlining specific logic inside of this function. There's already too much of that here already.

Reply:
okay. Once i fix the logic i will split it out into a new method

Reply:
i would have to cache the results of the logic in the function `is_divergent`, so that if i get `function >=0` for all x in integration limits while evaluating the `solve_univariate_inequality(f>=0,x)` i can use the result in `doit` to give the integration result as `oo/-oo` based on integration limits order rather than running this code again.